src/vec_buf_doubled.h

```diff
@@ -0,0 +1,69 @@
+/* -*- mode: c++ -*- */
+#ifndef __NTTEC_VEC_BUF_DOUBLED_H__
+#define __NTTEC_VEC_BUF_DOUBLED_H__
+
+#include "vec_buffers.h"
+
+namespace nttec {
+namespace vec {
+
+/** A vector of `n` buffers virtually extented to `2n`.
+ *
+ * Physically, BuffersDoubled contains `n` buffers but behaves as if it
+ * contained `2n` buffers.
+ *
+ * Example:
+ *
+ * We have a set of `N` independent buffers (+ a buffer of zeros):
```

Comment:
`(+ a buffer of zeros)` should not be there, shouldn't it?

Reply:
Good catch!

Too much copy/paste